src/token/Token.sol

```diff
@@ -50,14 +62,67 @@ contract Token is ERC20, ERC20Burnable, ERC20Permit, AccessControl {
         _mint(LIQUIDTY_ADDRESS, LIQUIDTY);
     }
 
-    /**
-    * @dev allow minting of tokens upto the MAX SUPPLY by MINTER
-    */
+    /// @dev allow minting of tokens upto the MAX SUPPLY by MINTER_ROLE
+    /// @param to is the address that will receive the minted tokens
+    /// @param amount is the amount of tokens to mint
     function mint(address to, uint256 amount) public onlyRole(MINTER_ROLE) {
         require(totalSupply() + amount <= MAX_SUPPLY, "Max supply reached");
         _mint(to, amount);
     }
+
+    /// @dev override the default transfer function to add tax if applicable
+    /// @param from is the address that will send the tokens
+    /// @param to is the address that will receive the tokens
+    /// @param amount is the amount of tokens to transfer
+    function _transfer(
+        address from,
+        address to,
+        uint256 amount
+    ) internal virtual override {
+        if (taxable && !isTransferFeeExempt(from)) {
+            _transferWithFees(from, to, amount);
+        } else {
+            super._transfer(from, to, amount);
+        }
+    }
+
+    /// @dev The function to transfer the fees applicable only if taxable is true
+    /// @param from is the address that will send the tokens
+    /// @param to is the address that will receive the tokens
+    /// @param amount is the amount of tokens to transfer
+    function _transferWithFees(
+        address from,
+        address to,
+        uint256 amount
+    ) internal {
+        uint256 tax = (amount * percentage) / 100;
+        uint256 netAmount = amount - tax;
+        super._transfer(from, to, netAmount);
+        super._transfer(from, FEE_ADDRESS, tax);
+    }   
+
+    /// @dev allow taxation of tokens by the owner
+    /// @param _taxable is the boolean to set if the token is taxable or not
+    /// @param _percentage is the percentage of tax to apply
+    function setTaxable(bool _taxable, uint8 _percentage) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        taxable = _taxable;
+        percentage = _percentage;
+    }
```

Comment:
Here we could add 2 checks
1. A require statement to ensure that _percentage lies in the range 0-100
2. (Optional) An if statement such that percentage is set only if _taxable is true